python/cudf/udf_cpp/groupby/function.cu

```diff
@@ -88,7 +69,8 @@ __device__ void device_sum(T const* data, int const items_per_thread, cudf::size
     }
   }
 
-  atomicAdds(sum, local_sum);
+  cuda::atomic_ref<T, cuda::thread_scope_device> ref{*sum};
```

Comment:
I have never used cuda::atomic_ref previously but shouldn't we use the cuda::thread_scope_block since we are only doing sum on the same threadblock?

Reply:
Great catch! Fixed.